### DISCUSSION TITLE
Add modification/creation dates to records

### DISCUSSION THREAD

**Discussion by jonasraoni** [pkp/pkp-lib#7977] [Proposals]
In order to improve debugging in general (knowing when something started/stopped happening is very useful) it would be great if all database records had a modifed_date/created_date.

**Cons**
- Increased database size
- Extra code to update the `modified_date`

**Solution**
- Through the database: MySQL has the `DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP`. PostgreSQL would need a trigger for the `modified_date`.
- Through the code: no big deal, but there's a higher probability of forgetting to update the modified date here and there.

**Comments**
- We can adopt this slowly.
- An extension of this would be an auditable database (keep all actions, what was modified and who did it), but I guess we don't have this need.

**Comment by NateWr** [comment]
We support this on some entities, with Laravel's `created_at` and `modified_at` columns. Do you have a list of other objects which should be treated this way? 

The `EntityDAO` pattern should make it very easy to track this -- however, we have to be careful to distinguish use cases for this data. Where it is implemented around submissions, now, we use it to track last activity. But this shouldn't be updated in cases like imported data or with data manipulations performed during migrations.

> An extension of this would be an auditable database (keep all actions, what was modified and who did it), but I guess we don't have this need.

Not a big need now, but something I'd like to support eventually. With our new event system, I'd like to fire an event with every add/edit/delete operation, which would allow a logging system to be built on top of the core code as needed.
